### DISCUSSION TITLE
Energy saving to to reduce battery consumption?

### DISCUSSION THREAD

**Discussion by StBoom** [klaasnicolaas/home-assistant-glow#202] [General]
Is there a way for energy saving?
Use the deep sleep, wake up via interrupt or similar?

I can unfortunately only use a power bank at the power meter and that is really fast empty.

**Comment by cieska** [comment]
My very random idea is to design something with a clamp transformer https://denkovi.com/split-core-current-transformer-with-0-5v-dc-output in this case you not interacting with live wires. But this on its own won't work. Need something more to it. Or maybe it can be as an additional supply to charge up batteries when power consumption is high? Well, as I said this is just a concept idea. 

Or you need two of those https://64.cdn.ekm.net/ekmps/shops/hardwarexpress/images/rec14-12-yuasa-battery-12v-14ah-2671-p.jpg?v=1
One on charge one in place. Will last for while until you need to change.

**Comment by StBoom** [comment]
I will brainstorm a bit. Let's see if I can convince my landlord.
Otherwise, it will be a thick battery.

**Reply by cieska** [reply]
Or run long cables for the Photodiode PCB. May need to bump up the power source as voltage can be dropped depends of length.

**Comment by squonk11** [comment]
I am also having the problem, that I don't have a supply voltage available at that location. Does anyone else have a good idea how to solve this issue? The big 12V battery is not a suitable solution for me; instead, a small power bank could be o.k..
I think the "Deep-Sleep" approach could be a good idea: each pulse wakes up the ESP32, then the time between the pulses is calculated and transmitted to HA via WiFi. During the deep sleep the WiFi is off in order to save energy.
